### PR TITLE
data-library: enable header interaction

### LINKED ISSUE #5944: data-library: implement header interaction
- filter by country bbox

### PR DESCRIPTION
**Pull request by matallo**
close #5944

it cannot be merged yet, as it is blocked by https://github.com/CartoDB/cartodb/issues/5961

CR @xavijam

### CODE REVIEW

**Comment by xavijam** [comment]
In terms of UI (cc @carlostallon):

- [ ] When a country is selected, map should center to that bounding box.

---

- [ ] When a country is so big, it chould hide header texts.

screen shot 2015-10-19 at 19 15 27

**Comment by xavijam** [comment]
@carlostallon please, could you review the UI?

File: lib/assets/javascripts/cartodb/data_library/header/template.jst.ejs [hunk 1]
```diff
@@ -5,14 +5,15 @@
...
+<div class="Header-inner Header-footer">
   <div class="CountrySelector js-CountrySelector u-inner">
     <p class="CountrySelector-text">Filter datasets by country selecting it in this map</p>
     <button class="CountrySelector-button js-country"></button>
   </div>
 
   <div class="CountrySelector CountrySelector-back js-CountrySelector-back is-hidden">
-    <div class="u-inner">
+    <div class="CountrySelector-inner u-inner">
+      <p class="CountrySelector-text">Back</p>
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/data_library/header/template.jst.ejs]: It should be a ```<button>``` right?

File: lib/assets/javascripts/cartodb/data_library/header/view.js [hunk 2]
```diff
@@ -52,33 +54,56 @@ var $ = require('jquery');
     };
 
     this.layer = L.geoJson(geojsonData, {
-      style: style
+      style: style,
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/data_library/header/view.js]: In my opinion this ```onEachFeature``` should be in a separate method, and that ```mouseover``` and ```mouseout``` too.

**Comment by carlostallon** [comment]
-  When a country is selected, map should center to that bounding box.
:+1: 

- When a country is so big, it chould hide header texts.
what about adding a gradient background to the header nav?

File: lib/assets/javascripts/cartodb/data_library/header/template.jst.ejs (hunk 1 above)
Comment (matallo) [inline lib/assets/javascripts/cartodb/data_library/header/template.jst.ejs]: the button is just a line below, that's the text accompanying it :D

File: lib/assets/javascripts/cartodb/data_library/header/view.js (hunk 2 above)
Comment (matallo) [inline lib/assets/javascripts/cartodb/data_library/header/view.js]: done

**Comment by matallo** [comment]
- added the background (I can see another alternative, which is removing the menu items)
- setview on bbox after click

**Pull request merged**